src/Features/Core/Portable/ChangeSignature/AbstractChangeSignatureService.cs

```diff
@@ -807,16 +812,35 @@ protected static int GetParameterIndex<TNode>(SeparatedSyntaxList<TNode> paramet
                             continue;
                         }
 
+                        SyntaxNode expression = null;
```

Comment:
whenever i see values set like this, and the possibly initialized, i think things would be a lot clearer by just extracting all of this into a helper.  i.e.:


```
var (expression, callsiteInferenceFailed) = await GenerateCallsiteExpressionAsync(...);
```

then the helper doesn't need complex data flow+assignment.  it can just trivially do what it needs, and return values it wants when appropriate.

Reply:
Done.